src/ncp/ncp_base_radio.cpp

```diff
@@ -451,11 +449,16 @@ otError NcpBase::HandlePropertySet_SPINEL_PROP_STREAM_RAW(uint8_t aHeader)
 
     if (error == OT_ERROR_NONE)
     {
-        // Don't do anything here yet. We will complete the transaction when we get a transmit done callback
+        // Cache the transaction ID for async response
+        mCurTransmitTID = SPINEL_HEADER_GET_TID(aHeader);
     }
     else
     {
-        error = WriteLastStatusFrame(aHeader, ThreadErrorToSpinelStatus(error));
+        // If we fail to report the error now, it will be reported later in HandleReceive() of ncp_base.cpp.
+        if (WriteLastStatusFrame(aHeader, ThreadErrorToSpinelStatus(error)) == OT_ERROR_NONE)
+        {
+            return OT_ERROR_NONE;
+        }
     }
```

Comment:
The above changes starting at line 443 are a bug we came across with how errors were being reported up.